mdagent/tools/base_tools/analysis_tools/rmsd_tools.py

```diff
@@ -2,314 +2,192 @@
-        self.path_registry = path_registry
-        self.pdb_file = self.path_registry.get_mapped_path(pdb)
-        self.trajectory = self.path_registry.get_mapped_path(traj)
-        self.ref_file = self.path_registry.get_mapped_path(ref)
-        self.ref_trajectory = self.path_registry.get_mapped_path(ref_traj)
-
-        # manually check for missing paths
-        if self.pdb_file == "Name not found in path registry.":
-            # set that file to None
-            self.pdb_file = None
-            self.pdb_name = None
-        else:
-            self.pdb_name = os.path.splitext(os.path.basename(self.pdb_file))[0]
-        if self.trajectory == "Name not found in path registry.":
-            self.trajectory = None
-        if self.ref_file == "Name not found in path registry." or self.ref_file is None:
-            self.ref_file = None
-            self.ref_name = None
-        else:
-            self.ref_name = os.path.splitext(os.path.basename(self.ref_file))[0]
-        if self.ref_trajectory == "Name not found in path registry.":
-            self.ref_trajectory = None
-        self.base_dir = "files"  # TODO: should update this to use checkpoint dir
-
-    def calculate_rmsd(
-        self,
-        rmsd_type="rmsd",
-        selection="backbone",
-        plot=True,
-    ):
-        if self.pdb_file is None:
-            raise FileNotFoundError("PDB file is required.")
-        self.filename = f"{rmsd_type}_{self.pdb_name}"
-
-        if rmsd_type == "rmsd":
-            if self.ref_file:
-                print("Calculating 1-D RMSD between two sets of coordinates...")
-                st.markdown(
-                    "Calculating 1-D RMSD between two sets of coordinates...",
-                    unsafe_allow_html=True,
-                )
-                return self.compute_rmsd_2sets(selection=selection)
-            else:
-                print("Calculating time-dependent RMSD...")
-                st.markdown(
-                    "Calculating time-dependent RMSD...", unsafe_allow_html=True
-                )
-                return self.compute_rmsd(selection=selection, plot=plot)
-        elif rmsd_type == "pairwise_rmsd":
-            print("Calculating pairwise RMSD...")
-            st.markdown("Calculating pairwise RMSD...", unsafe_allow_html=True)
-            return self.compute_2d_rmsd(selection=selection, plot_heatmap=plot)
-        elif rmsd_type == "rmsf":
-            print("Calculating root mean square fluctuation (RMSF)...")
-            st.markdown(
-                "Calculating root mean square fluctuation (RMSF)...",
-                unsafe_allow_html=True,
-            )
-            return self.compute_rmsf(selection=selection, plot=plot)
-        else:
-            raise ValueError(
-                "Invalid rmsd_type. Please choose from 'rmsd', 'pairwise_rmsd', 'rmsf'"
-            )
+    if traj_fileid is None:
+        print("Warning: no trajectory file provided.")
```

Comment:
This should either raise the warning, or give an error

Reply:
it's updated to a real warning when I moved to utils. Sometimes people just want to analyze PDB without trajectory file (rmsd can do this), so this won't raise an error.